### DISCUSSION TITLE
Mount issues/questions on Mac

### DISCUSSION THREAD

**Discussion by milyenpabo** [iterate-ch/cyberduck#14643] [Q&A]
I consider using MountainDuck as a replacement for SSHFS on my Mac client. My use case is to mount remote folders from Linux servers as local volumes on the Mac (using SFTP), and then edit the files via an editor running on the client.

From my first tests, it seems I can achieve this with MountainDuck, however, I see some issues:

1. When I use the terminal to `cd` into the mount folder on my Mac, the terminal session freezes (or sometimes the command succeeds, but it is very very slow). Seems like MountainDuck does not support visiting the mount folder from a terminal?

2. When I edit a file on the client, the changes are reflected on the sever pretty quickly. However, when I edit a file on the server, the changes are reflected on the client with varying latency, but it is usually slow (takes 10s of seconds). Is this a known issue, or can I configure the app to handle 2-way sync better? (I tried various config options from the MountainDuck GUI, but it did not help.)

I appreciate any help or feedback.

**Comment by dkocher** [comment]
Working with files from a shell in Terminal.app should work just fine.

1. When connected with _Smart Synchronization_ connect mode repeated directory listings should be instant. When connected with _Online_ connect mode this may depend on the latency to the server. 

2. When working on macOS, the directory listing will be cached. It may take up to 60 seconds for `ls` to return the latest contents from the server. See also our [documentation](https://docs.mountainduck.io/mountainduck/issues/#changes-from-server-not-immediately-visible).

**Reply by milyenpabo** [reply]
Thanks for the response. I'm testing Smart Synchronization now.

How about my second issue? It seems that sometimes my server side changes won't show up on the client, even for a long period of time. E.g. I'm waiting since 10 minutes for a server side change to be reflected on the client. Even if I disconnect + reconnect, the change is not reflected.

This is bad, as this can lead to data loss (the client can unknowingly overwrite a server side change).

Can this behavior be fixed somehow? Even a manual workaround is better than not knowing if I see the latest versions of the files on the client. E.g., I might try to switch off caching, but I don't see an option for that?

Any hints would be appreciated.

**Reply by dkocher** [reply]
> Even if I disconnect + reconnect, the change is not reflected.

The only explanation of such behaviour I have would be that the modification date on the folder does not implicitly change when its content changes which is what we assume from connections to SFTP servers. Please send a [debug log](https://docs.mountainduck.io/mountainduck/support/#logging-output) to [support](mailto:support@mountainduck.io).

**Reply by dkocher** [reply]
> This is bad, as this can lead to data loss (the client can unknowingly overwrite a server side change).

When synchronizing changes a conflict will be detected and the version already on the server will be moved away. See [Sync Conflicts](https://docs.mountainduck.io/mountainduck/sync/#sync-conflicts) in the documentation.

**Reply by milyenpabo** [reply]
Thank you for the response. I'm testing conflict management.

I did some more testing for syncing server side changes. I created some debug logs, I'll send it via mail.
